### PR TITLE
Adding new example using make edge function

### PR DESCRIPTION
**Pull request by reble**

### CODE REVIEW

File: sycl/test/graph/graph-explicit-saxpy.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    });
+  });
+
+  g.make_edge(init, compute);
+
+  auto executable_graph = g.finalize(q.get_context());
+
+  q.submit([&](sycl::handler &h) { h.exec_graph(executable_graph); });
+
+  auto result = y[0];
```
Comment (EwanC) [inline sycl/test/graph/graph-explicit-saxpy.cpp]: `result` variable isn't used, would either remove it or do some result verification on all of `y` (although it gets a bit involved)

E.g
```cpp
q.wait();

bool check = true;
for (size_t i = 0; i<n; i++) {
  if (y[i] != 5.0f) {
    check = false;
    break;
  }  
}
sycl::free(x, q);
sycl::free(y, q);

if (check)
  std::cout << "Passed\n";
else
  std::cout << "Failed\n";

return check ? 0 : -1;
```

**Review by EwanC — APPROVED** [review]
LGTM, minor comment

File: sycl/test/graph/graph-explicit-saxpy.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      size_t i = idx;
+      y[i] = a * x[i] + y[i];
+    });
+  });
+
+  g.make_edge(init, compute);
+
+  auto executable_graph = g.finalize(q.get_context());
+
+  q.submit([&](sycl::handler &h) { h.exec_graph(executable_graph); });
```
Comment (reble) [inline sycl/test/graph/graph-explicit-saxpy.cpp]:
```suggestion
  q.submit([&](sycl::handler &h) { h.exec_graph(executable_graph); }).wait();
```

File: sycl/test/graph/graph-explicit-saxpy.cpp (hunk 1 above)
Comment (reble) [inline sycl/test/graph/graph-explicit-saxpy.cpp]:
```suggestion
```

File: sycl/test/graph/graph-explicit-saxpy.cpp (hunk 2 above)
Comment (Bensuo) [inline sycl/test/graph/graph-explicit-saxpy.cpp]:
```suggestion
  q.submit([&](sycl::handler &h) { h.ext_oneapi_graph(executable_graph); });
```
To reflect recent changes.

File: sycl/test/graph/graph-explicit-saxpy.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+#include <CL/sycl.hpp>
```
Comment (Bensuo) [inline sycl/test/graph/graph-explicit-saxpy.cpp]:
```suggestion
#include <sycl/sycl.hpp>
```
I believe `CL/sycl.hpp` is deprecated in SYCL 2020 so we should prefer this in new tests I think.

File: sycl/test/graph/graph-explicit-saxpy.cpp (hunk 1 above)
Comment (Bensuo) [reply sycl/test/graph/graph-explicit-saxpy.cpp]: I think we should prefer having some simple verification for all examples rather than none.

**Review by Bensuo — APPROVED** [review]
LGTM just some minor comments.

File: sycl/test/graph/graph-explicit-saxpy.cpp (hunk 1 above)
Comment (reble) [reply sycl/test/graph/graph-explicit-saxpy.cpp]: Makes sense. I'll take the line out for now. The test can stay and makes sense as is. We can add verification of results and sanity checks later.

File: sycl/test/graph/graph-explicit-saxpy.cpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      size_t i = idx;
+      y[i] = a * x[i] + y[i];
+    });
+  });
+
+  g.make_edge(init, compute);
+
+  auto executable_graph = g.finalize(q.get_context());
+
+  q.submit([&](sycl::handler &h) { h.exec_graph(executable_graph); }).wait();
```
Comment (reble) [inline sycl/test/graph/graph-explicit-saxpy.cpp]:
```suggestion
  q.submit([&](sycl::handler &h) { h.ext_oneapi_graph(executable_graph); }).wait();
```

File: sycl/test/graph/graph-explicit-saxpy.cpp (hunk 1 above)
Comment (reble) [reply sycl/test/graph/graph-explicit-saxpy.cpp]: Created related issue #70

**Review by EwanC — APPROVED** [review]
LGTM, merge it

**Pull request merged**